// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

message GkehubBetaFeature {
  string name = 1;
  map<string, string> labels = 2;
  GkehubBetaFeatureSpec spec = 3;
  string create_time = 4;
  string update_time = 5;
  string delete_time = 6;
  string project = 7;
  string location = 8;
}

message GkehubBetaFeatureSpec {
  GkehubBetaFeatureSpecMulticlusteringress multiclusteringress = 1;
}

message GkehubBetaFeatureSpecMulticlusteringress {
  string config_membership = 1;
}

message ApplyGkehubBetaFeatureRequest {
  GkehubBetaFeature resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteGkehubBetaFeatureRequest {
  string service_account_file = 1;
  GkehubBetaFeature resource = 2;
}

message ListGkehubBetaFeatureRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListGkehubBetaFeatureResponse {
  repeated GkehubBetaFeature items = 1;
}

service GkehubBetaFeatureService {
  rpc ApplyGkehubBetaFeature(ApplyGkehubBetaFeatureRequest) returns (GkehubBetaFeature);
  rpc DeleteGkehubBetaFeature(DeleteGkehubBetaFeatureRequest) returns (google.protobuf.Empty);
  rpc ListGkehubBetaFeature(ListGkehubBetaFeatureRequest) returns (ListGkehubBetaFeatureResponse);
}
